### ISSUE TITLE
Defaults not working as expected for size-limited array

### ISSUE THREAD

**Issue by murrple-1** [bhch/django-jsonform#48]
Creating a form using this schema:
```python
DEFAULT_PARAMETERS_SCHEMA = {
    "type": "object",
    "properties": {
        "max_weekday_characteristics": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {},
                "additionalProperties": {
                    "type": "integer",
                    "helpText": "Key is a tag-name. Value is the maximum for that tag",
                },
            },
            "minItems": 7,
            "maxItems": 7,
            "default": [{}, {}, {}, {}, {}, {}, {}],
            "title": "Maximum Characteristics per Weekday",
        },
    },
}
```
has unexpected behaviour. I would expect that the default form would look like this:
![Screenshot from 2022-08-12 12-24-41](https://user-images.githubusercontent.com/5559656/184429190-be8ef693-da60-4742-b131-9298e4df023d.png)
but it instead defaults to being an empty list. I *think* this is a bug, but please advise.

**Comment by bhch** [comment]
I tested your schema and the form *does* look like the screenshot. Which version are you using? Check using this command:

```
$ python -c "import django_jsonform; print(django_jsonform.__version__)"
```

**Comment by murrple-1** [comment]
Using `2.10.1`

**Comment by bhch** [comment]
That is strange. Could you check browser console for errors? Also, is there any initial data being passed to the field?

---

BTW, I've created an interactive playground here: https://bhch.github.io/react-json-form/

Paste the following schema (I've removed trailing commas to make it valid JSON) in the textarea. You should see the form as expected. 

```json
{
    "type": "object",
    "properties": {
        "max_weekday_characteristics": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {},
                "additionalProperties": {
                    "type": "integer",
                    "helpText": "Key is a tag-name. Value is the maximum for that tag"
                }
            },
            "minItems": 7,
            "maxItems": 7,
            "default": [{}, {}, {}, {}, {}, {}, {}],
            "title": "Maximum Characteristics per Weekday"
        }
    }
}
```

**Comment by murrple-1** [comment]
Okay, some progress on my end.
Changing the field from
```python
default_parameters = JSONField(
        schema=DEFAULT_PARAMETERS_SCHEMA,
        default=dict,
        null=True,
        blank=True,
    )
```
to
```python
default_parameters = JSONField(
        schema=DEFAULT_PARAMETERS_SCHEMA,
    )
```
may have made some difference. Exploring now...

**Comment by bhch** [comment]
When you pass initial or default data  (e.g. through `default=dict`), the form widget tries to "sync" the data with the schema.

In this case, the initial data is an empty dict `{ }`. So, the form widget adds the `max_weekday_...` key to data and sets it to an empty array. The array is kept empty due to a previous bug report as adding blank items to arrays could be undesirable.

But I guess, in this case when you've explicitly set `minItems`, then the array should have at least those many items.

This is definitely a bug. 

Thank you for reporting this.

**Comment by bhch** [comment]
I've fixed this in the JavaScript code. It will be released in the next version.

However, I'd like to clear up a few things:

Case 1: If initial data for the whole field **not provided** (through field's `default` parameter):

 - Widget will first populate the data array with schema's `default` values.
 - If array is still less than `minItems`, then new items will be added to the array either from `items.default` or empty string or null (depending on type).

Case 2: If initial data for the shole field **is provided** (through field's `default` parameter):

 - Widget will try to "sync" the data with the schema but with certain restrictions. 
 - Widget will ignore the schema array's `default` keyword. This is because in the frontend there's no distinction between a field's default data or field's data saved in the database. Hence, we don't want to implicitly modify the field's data because what if the data is coming from database and user has already changed the default values previously. It would be unexpected if the widget will again add the default values to the data.
 - If array has `minItems`, only then new items will be added either from `items.default` or empty string or null (array's `default` is still ignored).

**Comment by bhch** [comment]
Fixed in [v2.11.0](https://github.com/bhch/django-jsonform/releases/tag/v2.11.0). Please upgrade.
